src/biocommons/seqrepo/dataproxy.py

```diff
@@ -0,0 +1,297 @@
+        """Translate given identifier to a list of identifiers in the
+        specified namespace.
+
+        `identifier` must be a string
+        `namespace` is case-sensitive
+
+        On success, returns string identifier.  Raises KeyError if given
+        identifier isn't found.
+
+        """
+
+        try:
+            md = self.get_metadata(identifier)
+        except (ValueError, KeyError, IndexError) as e:
+            raise KeyError(identifier) from e
+        aliases = list(set(md["aliases"]))  # ensure uniqueness
+        if namespace is not None:
+            nsd = namespace + ":"
+            aliases = [a for a in aliases if a.startswith(nsd)]
+        return aliases
+
+
+class _SeqRepoDataProxyBase(_DataProxy):
+    # wraps seqreqpo classes in order to provide translation to/from
+    # `ga4gh` identifiers.
+
+    def get_metadata(self, identifier):
+        md = self._get_metadata(identifier)
+        md["aliases"] = list(a for a in md["aliases"])
+        return md
+
+    def get_sequence(self, identifier, start=None, end=None):
+        return self._get_sequence(identifier, start=start, end=end)
+
+    @abstractmethod
+    def _get_metadata(self, identifier):  # pragma: no cover
+        pass
+
+    @abstractmethod
+    def _get_sequence(self, identifier, start=None, end=None):  # pragma: no cover
+        pass
+
+
+class SeqRepoDataProxy(_SeqRepoDataProxyBase):
+    """DataProxy based on a local instance of SeqRepo"""
+
+    def __init__(self, sr):
+        super().__init__()
+        self.sr = sr
+
+    def _get_sequence(self, identifier, start=None, end=None):
+        # fetch raises KeyError if not found
+        return self.sr.fetch_uri(coerce_namespace(identifier), start, end)
+
+    def _get_metadata(self, identifier):
+        ns, a = coerce_namespace(identifier).split(":", 2)
+        r = list(self.sr.aliases.find_aliases(namespace=ns, alias=a))
+        if len(r) == 0:
+            raise KeyError(identifier)
+        seq_id = r[0]["seq_id"]
+        seqinfo = self.sr.sequences.fetch_seqinfo(seq_id)
+        aliases = self.sr.aliases.find_aliases(seq_id=seq_id)
+        md = {
+            "length": seqinfo["len"],
+            "alphabet": seqinfo["alpha"],
+            "added": _isoformat(seqinfo["added"]),
+            "aliases": [f"{a['namespace']}:{a['alias']}" for a in aliases],
+            }
+        return md
+
+
+class SeqRepoRESTDataProxy(_SeqRepoDataProxyBase):
+    """DataProxy based on a REST instance of SeqRepo, as provided by seqrepo-rest-services"""
+
+    rest_version = "1"
+
+    def __init__(self, base_url):
+        super().__init__()
+        self.base_url = f"{base_url}/{self.rest_version}/"
+
+    def _get_sequence(self, identifier, start=None, end=None):
+        url = self.base_url + f"sequence/{identifier}"
+        _logger.info("Fetching %s", url)
+        params = {"start": start, "end": end}
+        resp = requests.get(url, params=params)
```

Comment:
a more helpful logging mechanism might be the URL + params being used; though i'll leave this here as a reminder to myself to look at some users of this library to tell whether they might already be doing that 🤔

Reply:
i looked and the existing caller in HGVS at least is not logging out what it's calling with, so i think it would be helpful to log the query params as well

Reply:
i updated a PR with the params needed to log here